### PR TITLE
Fix Factory visibility

### PR DESCRIPTION
**Pull request by cbruegg**
Since `FACTORY` was defined inside the companion object of the `internal` class `KotlinJsonAdapter`, it was inaccesible from Kotlin modules other than `moshi-kotlin`. I've fixed this by making it a top-level singleton `object`.

### CODE REVIEW

**Comment by cbruegg** [comment]
I wonder why this didn't trigger Travis CI as other PRs did?

File: kotlin/src/main/java/com/squareup/moshi/KotlinJsonAdapter.kt [hunk 1]
```diff
@@ -143,61 +144,61 @@ internal class KotlinJsonAdapter<T> private constructor(
       return if (value !== ABSENT_VALUE) value else null
     }
   }
+}
 
-  companion object {
-    @JvmField val FACTORY = Factory { type, annotations, moshi ->
-      if (!annotations.isEmpty()) return@Factory null
+object KotlinJsonAdapterFactory : JsonAdapter.Factory {
```
Comment (JakeWharton) [inline kotlin/src/main/java/com/squareup/moshi/KotlinJsonAdapter.kt]: This has ugly implications for Java callers. We need `@file:JvmName` to get a decent named enclosing class and then we want this to be exposed ideally as a `FACTORY` field.

Comment (cbruegg) [inline kotlin/src/main/java/com/squareup/moshi/KotlinJsonAdapter.kt]: It's accessed using `KotlinJsonAdapterFactory.INSTANCE` from Java in this PR.

Using `@file:JvmName("Something")` and naming it `FACTORY` would mean it's accessed like this from Java: `Something.FACTORY` and simply `FACTORY` from Kotlin. The latter seems to be worse than the downside of appearing as `KotlinJsonAdapterFactory.INSTANCE` from Java caused by this PR.

There's another option: Make `KotlinJsonAdapter` and thus the companion object public, but all of its non-inherited members `internal`. That may be error-prone though.

Comment (JakeWharton) [inline kotlin/src/main/java/com/squareup/moshi/KotlinJsonAdapter.kt]: Ah yeah it's a type not a function. I suppose that's an acceptable trade-off

**Comment by cbruegg** [comment]
Merged `master` into this to fix the build.

**Comment by swankjesse** [comment]
Thanks!

**Pull request merged**